frontend/src/routes/UploadProperty.jsx

```diff
@@ -273,6 +296,111 @@ const PropertyUpload = () => {
 
 
     
+    
+    const convertToManwon = () => {
+        if (!isConverted) {
+          // Store original prices
+          setOriginalPrices({
+            "보증금": propertyData.보증금,
+            "월세": propertyData.월세,
+            "전체 금액": propertyData.보증금 + propertyData.월세
+          });
+    
+          // Convert to 만원 (divide by 10,000)
+          const convertedData = {
+            "보증금": (propertyData.보증금 / 10000).toString(),
+            "월세": (propertyData.월세 / 10000).toString(),
+            "전체 금액": ((propertyData.보증금 + propertyData.월세) / 10000).toString()
+          };
+    
+          setPriceManwon(convertedData);
+          setIsConverted(true); // Mark as converted
+        } else {
+          // Restore original prices
+          setPriceManwon(originalPrices);
+          setIsConverted(false); // Mark as not converted
+        }
+      };
+    
+      const handleNewAddressSearch = async (e) => {
+        const searchText = e.target.value;
+        setNewAddress(searchText);  // Update local state for new address input
+      
+        if (searchText.length > 0) {
+          try {
+            const response = await axios.get(`http://localhost:8000/addresses?searchText=${searchText}`);
+            setNewAddressSuggestions(response.data);  // Set suggestions for new address
+            console.log(response.data);
+          } catch (error) {
+            console.error('Error fetching new address suggestions:', error);
+          }
+        } else {
+          setNewAddressSuggestions([]);  // Clear suggestions when input is less than 3 characters
+        }
+      };
+      
+      // Handle the selection of a new address
+      const handleNewAddressSelect = (selectedAddress) => {
+         // Format the new address by combining different parts
+         const formattedNewAddress = `${selectedAddress.new_district} ${selectedAddress.new_town} ${selectedAddress.new_road_name} ${selectedAddress.new_building_main_number}${selectedAddress.new_building_sub_number!=0 ? "-" + selectedAddress.new_building_sub_number : ""}`;
+         setNewAddress(formattedNewAddress);  // Auto-fill the corresponding new address
+
+         // Set the old address in the local state
+         const formattedOldAddress = `${selectedAddress.old_district} ${selectedAddress.old_town} ${selectedAddress.old_village} ${selectedAddress.old_lot_main_number}${selectedAddress.old_lot_sub_number!=0 ? "-" + selectedAddress.old_lot_sub_number : ""}`;
+         setOldAddress(formattedOldAddress);  //
+      
+         let addressId = selectedAddress.new_address_id        ;  // Scientific notation
```

Comment:
In MySQL, the storage size for `VARCHAR` and `DECIMAL(25, 0)` differs based on how they are represented and stored. Here's a comparison:

### 1. **`VARCHAR` Storage Size**
The storage size of a `VARCHAR(N)` is determined by:
- The number of characters stored.
- The character set used (e.g., UTF-8 uses 1 to 4 bytes per character).
- An additional 1 or 2 bytes for length information.

For example:
- **`VARCHAR(25)`** with ASCII/Latin1 encoding (1 byte per character) will use:
  - Up to 25 bytes for the characters.
  - 1 byte of length information (if length ≤ 255 characters).

Thus, **`VARCHAR(25)`** would take up to **26 bytes** in ASCII (25 bytes for the number and 1 byte for the length).

If you're using UTF-8 encoding (which can take up to 4 bytes per character), the storage would be up to **101 bytes** (25 characters * 4 bytes + 1 byte for length).

### 2. **`DECIMAL(25, 0)` Storage Size**
The storage size of a `DECIMAL(M, D)` in MySQL is based on the number of digits `M`:
- Each group of 9 digits requires 4 bytes.
- The remaining digits are stored proportionally.

For **`DECIMAL(25, 0)`**, this means:
- 25 digits split into groups of 9 digits: 2 full groups (18 digits) and 1 partial group (7 digits).
- Each group of 9 digits takes **4 bytes**, and the remaining 7 digits take **4 bytes** (since 1-9 digits always take 4 bytes).

Total size: 
- **(4 bytes + 4 bytes + 4 bytes) = 12 bytes**.

### Summary of Size Comparison:
- **`VARCHAR(25)`**: 
  - 26 bytes (with ASCII encoding, 1 byte per character).
  - Up to 101 bytes (with UTF-8 encoding).
  
- **`DECIMAL(25, 0)`**: 
  - 12 bytes fixed.

### Conclusion:
`DECIMAL(25, 0)` is significantly more space-efficient (12 bytes) compared to `VARCHAR(25)` (which can use 26 to 101 bytes depending on encoding). Use `DECIMAL(25, 0)` when storing a numeric value like a 25-digit integer for optimal space and performance.

Reply:
In JavaScript, for very large integers like a 25-digit integer, you should use the **`BigInt`** data type.

### Why `BigInt`?
- **`BigInt`** is specifically designed for handling large integers that exceed the precision limit of the regular `Number` type.
- A regular JavaScript `Number` uses 64-bit floating-point representation, which can safely represent integers only up to **`2^53 - 1`** (i.e., **`9,007,199,254,740,991`**). This is far smaller than a 25-digit number.
- **`BigInt`** can handle arbitrarily large integers, including 25-digit numbers and beyond, without losing precision.

### How to use `BigInt`:
You can declare a `BigInt` by appending an **`n`** to the end of the integer or by using the `BigInt()` constructor.

#### Example:
```javascript
const bigNumber = 1234567890123456789012345n; // Using 'n' notation
// OR
const bigNumber = BigInt("1234567890123456789012345"); // Using constructor
```

### Summary:
- **Use `BigInt`** for 25-digit integers in JavaScript to avoid precision issues.
- Regular `Number` in JavaScript cannot safely handle integers this large.